src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1240,12 +1235,8 @@ public bool StopAndAbort()
             aborted |= _sendQueue.StopAndAbort(this);
             aborted |= _receiveQueue.StopAndAbort(this);
 
-            lock (_registerLock)
-            {
-                // Freeing the token will prevent any future event delivery.  This socket will be unregistered
-                // from the event port automatically by the OS when it's closed.
-                _asyncEngineToken.Free();
-            }
+            // The handle is still valid and about to be closed.
+            _asyncEngine?.UnregisterSocket(_socket.DangerousGetHandle());
```

Comment:
What prevents a call to Register from racing with a call to StopAndAbort, such that _asyncEngine is set to non-null after this?

Reply:
This gets called as part of `ReleaseHandle`. Register can't do a successful`DangerousAddRef` at this point.
Should I make it more explicit in the comment?

Reply:
> Should I make it more explicit in the comment?

That'd be helpful, thanks.